### ISSUE TITLE
How to add russia country?

### ISSUE THREAD

**Issue by Azat868** [openalpr/openalpr#558]
I add rus.xml  in runtime_data/region
Create runtime_data/config/ru.conf and write rus.xml
After alpr -c ru path/image.jpg no lisensed plate. This does not work(
Please give me russian conf file( 

And what is osr_language = lus in .conf file? where russian language runtime_data/ocr/tessdata/lru.traineddata?
 Will this work https://github.com/KostyaKulakov/Russian_System_of_ANPR/blob/master/runtime_data/tessdata/amh.traineddata ?

Please help me=(

**Comment by Azat868** [comment]
ru.conf
`; One-line European style plates

; 35-50; 45-60, 55-70, 65-80, 75-90
char_analysis_min_pct = 0.35
char_analysis_height_range = 0.15
char_analysis_height_step_size = 0.10
char_analysis_height_num_steps = 5

segmentation_min_speckle_height_percent = 0.2
segmentation_min_box_width_px = 5
segmentation_min_charheight_percent = 0.4;
segmentation_max_segment_width_percent_vs_average = 2.0;

plate_width_mm = 520
plate_height_mm = 110

multiline = 0

char_height_mm = 80
char_width_mm = 53
char_whitespace_top_mm = 10
char_whitespace_bot_mm = 10

template_max_width_px = 184
template_max_height_px = 46

; Higher sensitivity means less lines
plateline_sensitivity_vertical = 18
plateline_sensitivity_horizontal = 55

; Regions smaller than this will be disqualified
min_plate_size_width_px = 65
min_plate_size_height_px = 18

; Results with fewer or more characters will be discarded
postprocess_min_characters = 5
postprocess_max_characters = 8

detector_file = ru.xml

ocr_language = amh

; Override for postprocess letters/numbers regex. 
postprocess_regex_letters = [А-X]
postprocess_regex_numbers = [0-9]


; Whether the plate is always dark letters on light background, light letters on dark background, or both
; value can be either always, never, or auto
invert = auto`

**Comment by x09** [comment]
@Azat868 
> detector_file = ru.xml

where to get this file?
откуда взять этот файл?

**Comment by ahvahsky2008** [comment]
@Azat868 @MarCovy  Получилось разобраться? А то я никак не могу конфиг пихнуть ему

**Comment by MarCovy** [comment]
@ahvahsky2008 can you write your comment in English, please ?

**Comment by ahvahsky2008** [comment]
@MarCovy, do you can add Russian language? Can you share solution?

**Comment by MarCovy** [comment]
Currently I am working at this. Now I have problem with training data and segmentation letters and numbers from the plates. Do you have any experince with them?

**Comment by ahvahsky2008** [comment]
@MarCovy, no (

**Comment by MarCovy** [comment]
@ahvahsky2008 I wrote him email. Maybe he join to our discussion.

**Comment by perchibald** [comment]
@MarCovy Do you need help with this ?? And how can I help?

**Comment by MarCovy** [comment]
@perchibald  thanks for your reply. For this moment I found the way how can I segment russian letters and numbers from licence plate. Picture will be below:
![image](https://user-images.githubusercontent.com/11411079/41714668-699d5b60-7551-11e8-8d68-ef65666123f8.png)
For this moment I must write some code to adjust my feature to whole code, but this is simple. 
The main problem for this moment is getting samples with russian plate for learning OCR. Do you have any set which can you share with me?

**Comment by perchibald** [comment]
@MarCovy I understand correctly? Need samples of russian car numbers in the picture format, jpg or png?

**Comment by MarCovy** [comment]
@perchibald  yes, you understand correctly :) samples can be in both format. This is not important, but  their size would be 250 px wide. This dimension applies to the licence plate, of course.

**Comment by MarCovy** [comment]
@perchibald yes, this will be perfect for me. How many?  Hmm, here is not upper limit. If I have a lot of samples, the ocr will be more accurate. Reversing the question, how many you can send me ?

**Comment by perchibald** [comment]
@MarCovy 
The latest clarifications! =) Do you need unique and real numbers? Or it is possible to use painted program ??
Like this:
![example1](https://user-images.githubusercontent.com/847633/41717698-f5cca718-7573-11e8-9c21-4134ad299586.png)
![example2](https://user-images.githubusercontent.com/847633/41717813-69ba2754-7574-11e8-8cd2-9db29f0cf3f5.png)

**Comment by MarCovy** [comment]
@perchibald 
So far I've used real photos. It showed good results, so I would prefer unique and real numbers on real photos, if it is not a problem.

**Comment by perchibald** [comment]
@MarCovy Got it! Give me a few days!

**Comment by MarCovy** [comment]
@perchibald 
Ok, can I ask for your email? Maybe I have others question, then I will send an email to you.

**Comment by perchibald** [comment]
@MarCovy Yes. 
perchibald@gmail.com

**Comment by ibraLab** [comment]
Народ, если возникли вопросы, то пишите мне попробуем вместе разобраться. 
Если у кого-то получилось, поделитесь пожалуйста!  
Мыло: ibrox9656@gmail.com

**Comment by AlexanderMatveev** [comment]
К черту мыло, давайте здесь 😊

**Comment by Azat868** [comment]
Как много времени прошло с тех пор. В общем шаблон я еще в том году сделал. Но пришел к выводу что это все таки игрушка.

Распознавать по фоткам авто одно дело. А когда ты на практике распознаешь это совсем другое. 

**Во первых** камера должна быть огромного разрешения, а на рынке, не то что на рынке, почти у всех стоят камеры 1920х1080. Смотрел тесты на одном сайте, так там ребята юзают 12 мегапискильную камеру. Карл, 12ти!  В основном успех распознавание зависит от разрешения. Когда смотришь 2 мегапискильное видео, оно красивое, вроде четкое, и ты думаешь ну а почему бы нет? Но как только ты нажимаешь стоп кадр, или ffmpeg сохраняешь кадры, то получается жуткое качество.

**Во вторых** камера должна быть расположена уровне номеров, чем ровнее и перпендикулярнеее тем лучше. На практике такое видел очень редко где. Угол снятого изображения играет огромную роль.

**В третьих** погодные условия. Я тестировал около 10к авто в светлое время суток, солнце, лето. Как только наступают сумерки, ночь, дождь и тд, распознавание сводится к 1% успеха. Нужны мощные ИК прожектора на эти цели, малое расстояние от камеры до номеров, что на практике не всегда удается.

В общем интересно было по играть с данной системой. Сначала радуешься как ребенок "вон вон смотри он даже марку авто распознал правильно", ну а потом понимаешь что это ни куда не годится. 
Шаб поищу на виртуалках, если кому надо выложу

**Comment by Azat868** [comment]
А ну про грязные номера и проистертую краску уж говорить не приходится, у 30% такие

**Comment by T5un4mI** [comment]
@Azat868 Конечно нужно, выложите пожалуйста. Очень любопытны результаты ваших исследований

**Comment by ibraLab** [comment]
@Azat868 как с вами можно связаться?

**Comment by RainGrid** [comment]
@MarCovy there are 2 bases with russian plates
https://yadi.sk/d/0H2AipxrcrXqy - uncropped
https://yadi.sk/d/EAfnQ947criHW - cropped
yavoloh@mail.ru - my email

**Comment by MarCovy** [comment]
@RainGrid 
I send you mail, check it.

**Comment by RainGrid** [comment]
https://github.com/RainGrid/openalpr_ru_runtime

**Comment by ramosjairo** [comment]
пожалуйста, кто может помочь мне читать русские номерные знаки.

**Comment by klim42** [comment]
hello, can you help me please

**Comment by RainGrid** [comment]
@klim42, what is Your question?

**Comment by klim42** [comment]
i install https://github.com/RainGrid/openalpr_ru_runtime but he does not recognize the plates

**Comment by klim42** [comment]
> @ klim42 , каков твой вопрос?

Установил конфиг по ссылке выше, не распознает...

**Comment by RainGrid** [comment]
> > @ klim42 , каков твой вопрос?
> 
> Установил конфиг по ссылке выше, не распознает...

Пока этот репозиторий просто для хранения нужных ресурсов, я несколько раз правил конфиг и, возможно, залил не лучшую версию. Однако даже с хорошим конфигом алгоритмы библиотеки плохо подходят под российские номера с частью региона. Нужно либо править исходники, либо придумать хитрый конфиг, либо использовать другие решения.

**Comment by kingstar788** [comment]
How to add iraq country???????????????????
